pandas/_libs/lib.pyx

```diff
@@ -1235,6 +1257,8 @@ cdef class Seen:
         bint datetimetz_      # seen_datetimetz
         bint period_          # seen_period
         bint interval_        # seen_interval
+        CurrSeen curr_seen    # Type of object we just saw
```

Comment:
Since there are individual members already on the class used for type-introspection, this seems pretty duplicative. Are the existing members not usable for the goal here? If not, we probably new a redesigned class structure for `Seen` instead of just adding this

Reply:
The existing members are for what is seen in the array as a whole, while this is just for the current element.

I will convert the existing members to the enum(probably going to rename it to ``SeenDtype``) when I start changing ``maybe_convert_objects`` (there are a couple extra ones, e.g. ``seen.datetimetz_`` that that function uses).